src/axolotl/utils/trainer.py

```diff
@@ -719,6 +720,9 @@ def setup_trainer(cfg, train_dataset, eval_dataset, model, tokenizer, total_num_
         **trainer_kwargs,
     )
 
+    LogPredictionCallback = log_prediction_callback_factory(trainer, tokenizer)
+    trainer.add_callback(LogPredictionCallback(cfg))
```

Comment:
@Glavin001 See if this could work the same as the one liner below:

```suggestion
    trainer.add_callback(log_prediction_callback_factory(trainer, tokenizer))
```

Reply:
No:

```
TypeError: log_prediction_callback_factory.<locals>.LogPredictionCallback.__init__() missing 1 required positional argument: 'cfg'
```